modules/secure-logging/slog.c

```diff
@@ -420,15 +424,34 @@ gchar *convertToBase64(unsigned char *input, gsize len)
  */
 void cmac(unsigned char *key, const void *input, gsize length, unsigned char *out, gsize *outlen)
 {
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
```

Comment:
Reflecting on
> It is in the secure-log module meaning that it is acceptable to directly interface with the OpenSSL API

Generally it is allowed to use OpenSSL API directly from any module. Just like before `tlscontext: move setup methods of DH-params to openssl-compat`, the `lib/tlscontext.c` file used them. Most of the functions in `lib/compat/openssl_support.c` are redefinitions of original OpenSSL functions for the sake of version compatibility, but sometimes we put "helper" functions there, just like you did in the mentioned commit.

I think what you did in that commit is really similar to moving out this function.

You might ask: what do we gain by moving these helper functions to the compat layer? I think it is better to see the OpenSSL version preprocessor `#if`s in one separate file, than them being everywhere in the codebase. So I would prefer moving these functions there.

Reply:
cmac() is not general enough in its current form (and is not well-audited) to be moved under compat/openssl.

We have to be really careful what we put there. IMHO the openssl_support unit should contain trivial copy-pated compatibility functions, other non-trivial helpers, but we should not place custom cryptographic primitives there unless they are general enough to be used anywhere and they are audited properly.